bash/scripts/get_actual_commit.sh

```diff
@@ -3,18 +3,32 @@
 set -eo pipefail
 
 get-actual-commit() {
+  repo_name="${1}"
   git_branch="${GIT_BRANCH:-$(git describe --all)}"
   git_commit="${GIT_COMMIT:-$(git rev-parse HEAD)}"
 
   if [[ "${git_branch}" != *"master"* ]]; then
-    # Determine actual PR commit for reporting status and Docker image tagging
-    git_commit_parents="${GIT_COMMIT_PARENTS:-$(echo "${git_commit}" | git log --pretty=%P -n 1 --date-order)}"
-
-    if [ "${#git_commit_parents}" -gt 40 ]; then
-      # More than one merge commit parent signifies that the merge commit is not the PR commit
-      git_commit="${git_commit_parents:0:40}"
-    # else only one merge commit parent signifies that the merge commit is also the PR commit
+    pr_commit="$(get-pr-commit "${repo_name}" "${git_commit}")"
+    if [ -n "${pr_commit}" ]; then
+      git_commit="${pr_commit}"
     fi
   fi
   echo "${git_commit}"
 }
+
+get-pr-commit() {
+  repo_name="${1}"
+  commit="${2}"
+
+  # if pr, commit will be the merge commit with pr commit and master commit as its parents
+  # curl GH api to derive actual pr commit for reporting status and Docker image tagging
+  resp=$(curl \
+  -sSL \
+  --user deis-admin:"${GITHUB_ACCESS_TOKEN}" \
+  "https://api.github.com/repos/deis/${repo_name}/commits/${commit}")
+
+  echo "${resp}" \
+    | docker run -i --rm kamermans/jq '.commit.message' \
```

Comment:
yea @helgi , have been doing that since jenkins slaves don't have jq yet, but do have docker.  

I can issue a follow-up pr in deis/ci-manager to get it on there and then refactor the containerized jq out of jenkins-jobs if desired.